presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/RandomNodeSelection.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler.nodeSelection;
+
+import com.facebook.presto.execution.RemoteTask;
+import com.facebook.presto.execution.scheduler.NodeMap;
+import com.facebook.presto.execution.scheduler.ResettableRandomizedIterator;
+import com.facebook.presto.metadata.InternalNode;
+import com.facebook.presto.metadata.Split;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.facebook.presto.execution.scheduler.NodeScheduler.randomizedNodes;
+import static com.facebook.presto.execution.scheduler.NodeScheduler.selectNodes;
+import static com.google.common.base.Verify.verify;
+import static com.google.common.collect.Sets.newHashSet;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class RandomNodeSelection
+        implements NodeSelection
```

Comment:
I don't see a need to make this abstract anymore. There is only one implementation.